arangod/Agency/Store.cpp

```diff
@@ -135,6 +136,7 @@ Store& Store::operator=(Store const& rhs) {
 Store& Store::operator=(Store&& rhs) {
   if (&rhs != this) {
     MUTEX_LOCKER(otherLock, rhs._storeLock);
+    MUTEX_LOCKER(lock, _storeLock);
```

Comment:
was this missing in general or was there a change such that this lock becomes necessary. If the later is true, where is the corresponding lock?

Reply:
This was missing as far as I can tell.

Reply:
Have you checked that we do not hold the lock at the caller's site?

Reply:
I have ran tests don't see anywhere.